src/stan/command/stanc_helper.hpp

```diff
@@ -228,6 +229,35 @@ inline int stanc_helper(int argc, const char* argv[],
     std::vector<std::string> include_paths;
     include_paths.push_back("");
 
+    if (cmd.has_key("include_paths")) {
+      std::string extra_path_str;      
+      cmd.val("include_paths", extra_path_str);
+
+      // extra_path_els is given explicitly so that multiple quote
+      // characters (in this case single and double quotes) can be
+      // used.
+      boost::escaped_list_separator<char> extra_path_els("\\", // escape char
+                                                         ",",  // delimiter
+                                                         "\"'" // quote chars
+                                                         );
+      
+      boost::tokenizer<
+        boost::escaped_list_separator<char>
+        > extra_path_tokenizer(extra_path_str, extra_path_els);
+
+      std::cout << "Include paths:\n";
```

Comment:
why is this printing to `std::cout`?  is this just to check your work during development?

Reply:
Nothing should ever print to `std::cout` in `stan-dev/stan`.  This causes undefined behavior in our interfaces.  Instead, if printing is really required, then it needs to use an argument output stream that will be configured higher up.

Reply:
the include path names are used in file `stan/io/program_reader.hpp`.   as it stands, the `read` method (https://github.com/stan-dev/stan/blob/7206bd9b0e2d5d585f54e0a8c1e14efca5f1350b/src/stan/io/program_reader.hpp#L219) searches all include paths in turn - lines 255-270.  if the include paths are bad, it will report and error - line 272.  to detect bad include paths, you'd need to change the code so that there's a record of the paths that were tried which can then be dumped into the error message.

Reply:
correct - if there are paths, add to message.

Reply:
Just FYI, unit tests have been added, and I followed the recommendation to only list the include paths if there is an error.